server/src/request.rs

```diff
@@ -65,52 +92,60 @@ pub fn process_analysis_request(request: AnalysisRequest) -> AnalysisResponse {
     for rule in &server_rules_to_rules {
         if !rule.verify_checksum() {
-            eprintln!("Rule {} has an invalid checksum", rule.name);
+            tracing::info!(
+                "Validation error: request rule `{}` has invalid checksum",
+                rule.name
+            );
             return AnalysisResponse {
                 rule_responses: vec![],
                 errors: vec![ERROR_CHECKSUM_MISMATCH.to_string()],
             };
         }
     }
 
+    let rules_count = rules.len();
+    let rules_str = if rules_count == 1 { "rule" } else { "rules" };
+    let rules_list = rules
+        .iter()
+        .map(|r| r.name.as_str())
+        .collect::<Vec<&str>>()
+        .join(", ");
     // execute the rule. If we fail to convert, return an error.
-    match rules_converted {
-        Ok(rules) => {
-            let rule_results = analyze(
-                &request.language,
-                rules,
-                &request.filename,
-                code_decoded_attempt.unwrap().as_str(),
-                &AnalysisOptions {
-                    use_debug: false,
-                    log_output: request
-                        .options
-                        .map(|o| o.log_output.unwrap_or(false))
-                        .unwrap_or(false),
-                },
-            );
+    let rule_results = analyze(
+        &request.language,
+        rules,
+        &request.filename,
+        code_decoded_attempt.unwrap().as_str(),
+        &AnalysisOptions {
+            use_debug: false,
+            log_output: request
+                .options
+                .map(|o| o.log_output.unwrap_or(false))
+                .unwrap_or(false),
+        },
+    );
 
-            let rule_responses = rule_results
-                .iter()
-                .map(|rr| RuleResponse {
-                    identifier: rr.rule_name.clone(),
-                    violations: rr.violations.iter().map(violation_to_server).collect(),
-                    errors: rr.errors.clone(),
-                    execution_error: rr.execution_error.clone(),
-                    output: rr.output.clone(),
-                    execution_time_ms: rr.execution_time_ms,
-                })
-                .collect();
+    let rule_responses = rule_results
+        .iter()
+        .map(|rr| RuleResponse {
+            identifier: rr.rule_name.clone(),
+            violations: rr.violations.iter().map(violation_to_server).collect(),
+            errors: rr.errors.clone(),
+            execution_error: rr.execution_error.clone(),
+            output: rr.output.clone(),
+            execution_time_ms: rr.execution_time_ms,
+        })
+        .collect();
 
-            AnalysisResponse {
-                rule_responses,
-                errors: vec![],
-            }
-        }
-        Err(_) => AnalysisResponse {
-            rule_responses: vec![],
-            errors: vec![ERROR_DECODING_BASE64.to_string()],
-        },
```

Comment:
Related to the comment above, but is this error handler needed

Reply:
line 74